src/EngineManager.php

```diff
@@ -59,6 +65,31 @@ protected function ensureAlgoliaClientIsInstalled()
         throw new Exception('Please install the Algolia client: algolia/algoliasearch-client-php.');
     }
 
+    /**
+     * Set the default Algolia config headers.
+     *
+     * @return array
+     */
+    protected function defaultAlgoliaHeaders()
+    {
+        if (! config('scout.user')) {
+            return [];
+        }
+
+        $headers = [];
+        $request = $this->container->make(Request::class);
+
+        if (filter_var($ip = $request->ip(), FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE)) {
+            $headers['X-Forwarded-For'] = $ip;
```

Comment:
I think its cool if you run some tests on Algolia dashboard. Like the same userToken, but different IPs shouldn't be consider a different user. You need to test this.